pkg/scheduler/watchers/membercluster/controller.go

```diff
@@ -169,10 +169,9 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 			return false
 		},
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			// Ignore deletion events; the removal of a cluster is first signaled by the join
-			// state change (join -> leave), which is already handled separately.
-			klog.V(2).Info("Ignoring delete events for member cluster objects", "eventObject", klog.KObj(e.Object))
-			return false
+			// the cluster is leaving the fleet
+			klog.V(2).Info("delete events for member cluster objects", "eventObject", klog.KObj(e.Object))
+			return true
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
```

Comment:
Hi Ryan! For this one we also need to enqueue when the deletion timestamp is added (updated) to the new object; the system considers the deletion op as an update event first (ObjectMeta updated) then a delete event (object removed).